dvc/utils/checksum.py

```diff
@@ -0,0 +1,147 @@
+"""Helpers for other modules."""
+
+from __future__ import unicode_literals
+
+from dvc.utils.compat import str, builtin_str, open
+
+import os
+import json
+import hashlib
+import logging
+import schema
+
+
+logger = logging.getLogger(__name__)
+
+LOCAL_CHUNK_SIZE = 1024 * 1024
+LARGE_FILE_SIZE = 1024 * 1024 * 1024
+
+CHECKSUM_MD5 = "md5"
+CHECKSUM_SHA256 = "sha256"
+CHECKSUM_MAP = {CHECKSUM_MD5: hashlib.md5, CHECKSUM_SHA256: hashlib.sha256}
+
+
+CHECKSUM_LOCAL_SCHEMA = {
+    schema.Optional(CHECKSUM_MD5): schema.Or(str, None),
+    schema.Optional(CHECKSUM_SHA256): schema.Or(str, None),
+}
+
+# NOTE: currently there are only 4 possible checksum names:
+#
+#    1) md5 (LOCAL, SSH, GS);
+#    2) etag (S3);
+#    3) checksum (HDFS);
+#    4) sha256 (LOCAL);
+#
+# so when a few types of outputs share the same name, we only need
+# specify it once.
+CHECKSUM_ETAG = "etag"
+CHECKSUM_CHECKSUM = "checksum"
+
+CHECKSUM_SCHEMA = CHECKSUM_LOCAL_SCHEMA.copy()
+CHECKSUM_SCHEMA[schema.Optional(CHECKSUM_ETAG)] = schema.Or(str, None)
+CHECKSUM_SCHEMA[schema.Optional(CHECKSUM_CHECKSUM)] = schema.Or(str, None)
+
+LOCAL_SUPPORTED_CHECKSUM_TYPES = [CHECKSUM_MD5, CHECKSUM_SHA256]
+HTTP_SUPPORTED_CHECKSUM_TYPES = [CHECKSUM_ETAG, CHECKSUM_MD5]
+SSH_SUPPORTED_CHECKSUM_TYPES = [CHECKSUM_MD5]
+AZURE_SUPPORTED_CHECKSUM_TYPES = [CHECKSUM_ETAG]
+GS_SUPPORTED_CHECKSUM_TYPES = [CHECKSUM_MD5]
+HDFS_SUPPORTED_CHECKSUM_TYPES = [CHECKSUM_CHECKSUM]
+S3_SUPPORTED_CHECKSUM_TYPES = [CHECKSUM_ETAG]
```

Comment:
Shouldn't this go to appropriate modules and just stay contained there? This will eliminate one level of indirection - these constants.

Reply:
They will go to the appropriate modules for sure 👍